### PR TITLE
[IMP] mrp: make mrp_bom_structure_report more extensible

### PR DESCRIPTION
**Pull request by mpanarin**
Description of the issue/feature this PR addresses:
make mrp_bom_structure_report more extensible 
by moving child values to a separate method. This way, other modules can
properly add or remove fields to the report without overriding it
completely and breaking the inheritance

Current behavior before PR:
The inner recursive method is not inheritable

Desired behavior after PR is merged:
it should be inheritable

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/mrp/report/mrp_bom_structure_report.py [hunk 1]
```diff
@@ -7,31 +7,42 @@
 class BomStructureReport(models.AbstractModel):
     _name = 'report.mrp.mrp_bom_structure_report'
 
+    @staticmethod
+    def _get_child_vals(record, level, qty, uom):
```
Comment (pedrobaeza) [inline addons/mrp/report/mrp_bom_structure_report.py]: What is `record` (which model)? A better argument name or a docstring in the method is desirable here for proper self-documentation

Comment (mpanarin) [reply addons/mrp/report/mrp_bom_structure_report.py]: @pedrobaeza updated

**Comment by yvaucher** [comment]
@sle-odoo ping

**Comment by mpanarin** [comment]
@sle-odoo ping )

**Comment by mpanarin** [comment]
@sle-odoo @pimodoo @fpodoo

**Comment by jgrandguillaume** [comment]
Hi @sle-odoo ,

Would you think you can see to merge this one ?

Thanks,

Joël

File: addons/mrp/report/mrp_bom_structure_report.py [hunk 2]
```diff
@@ -7,36 +7,54 @@
 class BomStructureReport(models.AbstractModel):
     _name = 'report.mrp.mrp_bom_structure_report'
 
-    def get_children(self, object, level=0):
+    @staticmethod
```
Comment (jco-odoo) [inline addons/mrp/report/mrp_bom_structure_report.py]: staticmethod?

**Comment by jco-odoo** [comment]
Merged in daa8fb62f745212752bdd7903029f96574f5be3a   We might have better reports in the next version.

**Comment by pedrobaeza** [comment]
@jco-odoo you have reference a wrong commit, so I understand this has been done in master, isn't it? It's not possible in v11?

**Comment by jco-odoo** [comment]
Hm, yes but it is in v11: https://github.com/odoo/odoo/commit/4178684fa371ab016518c107b30b72463c941b76

**Comment by pedrobaeza** [comment]
Great, thanks!

**Pull request closed without merging**